src/coordsim/simulation/flowsimulator.py

```diff
@@ -41,40 +41,41 @@ def generate_flow(self, node_id):
         Generate flows at the ingress nodes.
         """
         while True:
-            self.total_flow_count += 1
+            if self.params.inter_arr_mean[node_id] is not None:
```

Comment:
Did you test this with a trace where an ingress is set to None? Just to make sure it doesn't get stuck in this empty endless loop.